WebService/v1/Controllers/SimulationsController.cs

```diff
@@ -18,34 +21,69 @@ namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Controller
     public class SimulationsController : Controller
     {
         private const int MAX_DELETE_DEVICES = 100;
+
         private readonly ISimulations simulationsService;
+        private readonly IServicesConfig servicesConfig;
+        private readonly IDeploymentConfig deploymentConfig;
         private readonly IIotHubConnectionStringManager connectionStringManager;
+        private readonly IPreprovisionedIotHub preprovisionedIotHub;
         private readonly ISimulationAgent simulationAgent;
-
+        private readonly ISimulationRunner simulationRunner;
+        private readonly IRateLimiting rateReporter;
         private readonly ILogger log;
 
         public SimulationsController(
             ISimulations simulationsService,
+            IServicesConfig servicesConfig,
+            IDeploymentConfig deploymentConfig,
             IIotHubConnectionStringManager connectionStringManager,
+            IPreprovisionedIotHub preprovisionedIotHub,
             ISimulationAgent simulationAgent,
+            ISimulationRunner simulationRunner,
+            IRateLimiting rateReporter,
             ILogger logger)
         {
             this.simulationsService = simulationsService;
+            this.servicesConfig = servicesConfig;
+            this.deploymentConfig = deploymentConfig;
             this.connectionStringManager = connectionStringManager;
+            this.preprovisionedIotHub = preprovisionedIotHub;
             this.simulationAgent = simulationAgent;
+            this.simulationRunner = simulationRunner;
+            this.rateReporter = rateReporter;
             this.log = logger;
         }
 
         [HttpGet]
         public async Task<SimulationListApiModel> GetAsync()
         {
-            return new SimulationListApiModel(await this.simulationsService.GetListAsync());
+            var simulationList = await this.simulationsService.GetListAsync();
+            var simulationListApiModel = new SimulationListApiModel();
+            foreach (var x in simulationList)
```

Comment:
could all this work be done in the `SimulationListApiModel` class?

Reply:
looks like all the work done here is mapping from one layer to another, mapping tasks that usually have in the model classes